external/fv3fit/fv3fit/train_microphysics.py

```diff
@@ -286,17 +291,17 @@ def main(config: TrainConfig, seed: int = 0):
         config.test_url, config.nfiles_valid, config.model_variables
     )
 
-    train_set = next(iter(train_ds.shuffle(1_000_000).batch(50_000)))
+    if config.shuffle_buffer_size is not None:
+        train_ds = train_ds.shuffle(config.shuffle_buffer_size)
```

Comment:
Shuffle buffer over samples really seems to be detrimental to memory usage and training performance.  If using dict code, we can shuffle on the file names (gives us re-ordering every epoch) and then get a psuedo shuffle if we use the dataset `.interleave`.  I'm wondering if we can get away with it and not see a degradation in final training loss.   I'll prepare an update in a PR with some of the timing and memory usage comparisons.

Reply:
Seems like a good thing to try in a follow-up PR. The computational cost is probably more important for dense models. RNNs bottleneck on the actual network ops. My null hypothesis is that some degree of interleaving will be important for performance, but maybe proper shuffling doesn't matter.